core/src/main/java/org/mskcc/cbio/portal/dao/DaoGenePanel.java

```diff
@@ -181,4 +198,103 @@ public static void deleteGenePanel(GenePanel genePanel) throws DaoException {
             JdbcUtil.closeAll(DaoGenePanel.class, con, pstmt, null);
         }
     }
+
+    /**
+     * Updates an existing gene panel to the incoming set of genes.
+     *
+     * @param genePanel
+     * @param incoming
+     * @throws DaoException
+     */
+    public static void updateGenePanel(GenePanel genePanel, Set<CanonicalGene> incoming) throws DaoException {
+        String stableId = genePanel.getStableId();
+        Integer internalId = genePanel.getInternalId();
+
+        if (stableId == null) {
+            throw new DaoException("Gene Panel stable ID cannot be null.");
+        }
+
+        updateGenePanelGeneList(internalId, incoming);
+    }
+
+    /**
+     * Prints the genes to be added and removed from a gene panel in the database.
+     *
+     * @param genePanel
+     * @param incoming
+     * @throws DaoException
+     */
+    public static void updatePreview(GenePanel genePanel, Set<CanonicalGene> incoming) throws DaoException {
+        Pair pair = getAddRemove(incoming, extractGenePanelGenes(genePanel.getInternalId()));
```

Comment:
According to the [Java 1.5 documentation](https://docs.oracle.com/javase/1.5.0/docs/guide/language/static-import.html), one can call static methods without their class name. I didn't know this since IntelliJ suggested this import style.